components/benefit_sponsors/app/domain/benefit_sponsors/operations/benefit_applications/reinstate.rb

```diff
@@ -111,7 +111,13 @@ def reinstate_census_employees
             )
 
             enrollments.select {|enr| enr.terminated_on.blank? || enr.terminated_on == application_term_date }.each do |hbx_enrollment|
-              effective_on = is_retroactive_canceled ? hbx_enrollment.effective_on : @reinstate_on
+              effective_on = if is_retroactive_canceled
+                               hbx_enrollment.effective_on
+                             elsif hbx_enrollment.effective_on > @reinstate_on
+                               hbx_enrollment.effective_on
```

Comment:
The code seems to assign the same value to effective_on in two cases: when is_retroactive_canceled is true, and when hbx_enrollment.effective_on is greater than @reinstate_on. Is this intentional? If so, you could simplify the code by using a logical OR operator (||) to combine the two conditions. For example:
```
effective_on = if is_retroactive_canceled || hbx_enrollment.effective_on > @reinstate_on
                                hbx_enrollment.effective_on
                              else
                                @reinstate_on
                              end
 ```